pkg/operator/configobservation/apiserver/observer_apiserver.go

```diff
@@ -0,0 +1,199 @@
+		}
+		if err := unstructured.SetNestedField(previouslyObservedConfig, currentKeyFile, keyFilePath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	observedConfig := map[string]interface{}{}
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("apiservers.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	servingCertSecretName := apiServer.Spec.ServingCerts.DefaultServingCertificate.Name
+
+	if len(servingCertSecretName) == 0 {
+		return observedConfig, nil
+	}
+
+	certFile := fmt.Sprint("/etc/kubernetes/static-pod-resources/secrets/user-serving-cert/tls.crt")
+	if err := unstructured.SetNestedField(observedConfig, certFile, certFilePath...); err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+	keyFile := fmt.Sprint("/etc/kubernetes/static-pod-resources/secrets/user-serving-cert/tls.key")
+	if err := unstructured.SetNestedField(observedConfig, keyFile, keyFilePath...); err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	destinationLocation := resourcesynccontroller.ResourceLocation{
+		Name:      "user-serving-cert",
+		Namespace: operatorclient.OperatorNamespace,
+	}
+	sourceLocation := resourcesynccontroller.ResourceLocation{
+		Name:      servingCertSecretName,
+		Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace,
+	}
+	if err = listers.ResourceSyncer().SyncConfigMap(destinationLocation, sourceLocation); err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	return observedConfig, errs
+}
+
+func ObserveNamedCertificates(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	namedCertificatesPath := []string{"servingInfo", "namedCertificates"}
+
+	previouslyObservedConfig := map[string]interface{}{}
+	if currentNamedCertificates, _, err := unstructured.NestedSlice(existingConfig, namedCertificatesPath...); len(currentNamedCertificates) > 0 || err != nil {
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if err := unstructured.SetNestedSlice(previouslyObservedConfig, currentNamedCertificates, namedCertificatesPath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	observedConfig := map[string]interface{}{}
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("apiservers.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	namedCertificates := apiServer.Spec.ServingCerts.NamedCertificates
+	if len(namedCertificates) > 10 {
+		err := fmt.Errorf("apiservers.config.openshift.io/cluster: spec.servingCerts.namedCertificates cannot have more than 10 entries")
```

Comment:
@mfojtik we can choose the number, but this simplifies some assumptions for syncing things with multiplicity > 1.  other ideas welcome.